### PR TITLE
Remove unneeded persistence flags.

### PR DESCRIPTION
**Pull request by NathanKell**
RepairConfig will need its own rework

### CODE REVIEW

File: TestFlightFailure_IgnitionFail.cs [hunk 1]
```diff
@@ -12,9 +12,9 @@ namespace TestFlight
 {
     public class TestFlightFailure_IgnitionFail : TestFlightFailure_Engine
     {
-        [KSPField(isPersistant=true)]
+        [KSPField]
         public bool restoreIgnitionCharge = false;
-        [KSPField(isPersistant=true)]
+        [KSPField]
```
Comment (jwvanderbeck) [inline TestFlightFailure_IgnitionFail.cs]: `ignorePressureOnPad` should just be removed completely

File: TestFlightCore/TestFlightCore/TestFlightCore.cs [hunk 2]
```diff
@@ -40,9 +40,9 @@ public class TestFlightCore : PartModuleExtended, ITestFlightCore
         public float lastMET;
         [KSPField(isPersistant=true)]
         public bool initialized = false;
-        [KSPField(isPersistant=true)]
+        [KSPField]
         public float maxData = 0f;
-        [KSPField(isPersistant=true)]
```
Comment (jwvanderbeck) [inline TestFlightCore/TestFlightCore/TestFlightCore.cs]: `maxData` is calculated from the reliability modules on the part, and if not persisted will have to be recalculated after load.  Which isn't necessarily an issue if preferred, but does take more than 0 time to complete. 

See: https://github.com/KSP-RO/TestFlight/blob/master/TestFlightCore/TestFlightCore/TestFlightCore.cs#L135

**Comment by NathanKell** [comment]
Concur, it's never used, but I was narrow-targeting the PR. :)

**Comment by jwvanderbeck** [comment]
> Concur, it's never used, but I was narrow-targeting the PR. :)

Fair enough and certainly good practice.

**Comment by NathanKell** [comment]
Sorry, net fail. That meant to reply to the first. Re the second (maxData) my assumption was that if the base part cfg changed, i.e. changing the max data possible, then this should update--and it won't if this is marked persistent. Which is the more desired behavior?

**Comment by jwvanderbeck** [comment]
Configured (IE config node) maxData can't change at runtime.  What can change is a max data multiplier can be set on the core, so say something like KCT wants to make it so that you can get some data in a sim but max out at 30% of normal, it can apply a 0.3 modifier to maxData.

Configured MaxData is stored on the Reliability modules in the .CFG.  But it is actually the Core who is responsible for controlling that data follows that, so the core essentially caches the data from the reliability module.  The first time it is needed it grabs it from the reliability module and caches it, then uses the cached value thereafter.  That value specifically should never change at runtime.  IE it should work the same whether it is persisted or not, with the only difference being that it will have to be recached after a load if not.  Since it is a computed value I think I agree with you and would just let it get recached, so i'm good with the change.  I guess it was more just wanting to point it out.

**Pull request merged**